pkg/odo/cli/component/create.go

```diff
@@ -766,9 +766,21 @@ func (co *CreateOptions) downloadProject() error {
 		return errors.Errorf("Project type not supported")
 	}
 
-	err = util.GetAndExtractZip(zipUrl, path)
-	if err != nil {
-		return err
+	if project.Source.SparseCheckoutDir != nil && *project.Source.SparseCheckoutDir != "" {
```

Comment:
this code https://github.com/openshift/odo/pull/3042/files#diff-58fe314b4e3f9120402717ab8c2b3049R769-R783 can be extracted into a function, would make things cleaner